### PR TITLE
added solution of leetcode problem no:51

### PR DESCRIPTION
**Pull request by khushijain3103**
<hr>

## Domain of Contribution 📊

- [x ] Data Structures
- [x ] Algorithms
<hr>

## Checklist ✅

- [x ] I have added Comments and added the time and space complexity of the code (For DSA).
- [x ] I have performed a self-review of my own code or work.
- [x ] I have commented my code, particularly in hard-to-understand areas.
- [x ] I have checked my code and corrected any misspellings

<hr>

image
image
image

<hr>
#DESCRIPTION: We have to place queens on a chessboard such that no two queens can attack each other.
Basically we will find safe ways to place our queen.

#APPROACH: Did this question through BACKTRACKING.
We will just move on our board and check if it is safe to place queen here through a function and now we will move on next row if we are not able to place our queen there then we will backtrack and replace the position of previous queen.
<hr>

### CODE REVIEW

**Comment by PranavViswanathan** [comment]
this is great, but can you add the question as comments at the top of the code?

**Comment by khushijain3103** [comment]
Yes ofcourse

**Comment by PranavViswanathan** [comment]
Awesome, merging PR
